### PR TITLE
[in_app_purchase]Update json_annotation version

### PR DESCRIPTION
**Pull request by includecmath**
## Description

plugin in_app_purchase use low version of json_annotation, conflict with flutter_tools from sdk. So update version of json_annotation, resolve conflicts with other packages.

## Related Issues

Because in_app_purchase >=0.1.0 depends on json_annotation ^2.0.0 and every version of flutter_tools from sdk depends on json_annotation 3.0.0, in_app_purchase >=0.1.0 is incompatible with flutter_tools from sdk.
So, because plugins_gallery depends on both flutter_tools any from sdk and in_app_purchase 0.2.2+2, version solving failed.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by ened** [comment]
hi @includecmath again!
Thank you for this one - could you please clarify if the plugin, with your change, will still compile & work on the current stable branch? 
Also, please also sync with the latest master, increment the version & add a entry to the change log. That will help others to spot your change quickly.

Thank you once again

**Comment by ened** [comment]
One more thing - after changing the json_* dependencies in `dev_dependencies` we need to re-run the build runner command, and those re-generated files should be part of this PR. 

```
flutter packages pub run build_runner build --delete-conflicting-outputs
```

json_serializable had breaking changes rgd. nullable fields from v 2 to 3. I can't see things from a glimpse either, but I'm 99% sure the generated files changed.

**Comment by includecmath** [comment]
Thanks for reminding re-generated *.g.dart file @ened and the command should be:😂

`flutter pub run build_runner build`

**Comment by includecmath** [comment]
@cyanglaz , can you please review this?

**Comment by includecmath** [comment]
@cyanglaz Ok thank you for reminding, I raise the min sdk bound reference [json_serializable](https://raw.githubusercontent.com/dart-lang/json_serializable/master/example/pubspec.yaml)

File: packages/in_app_purchase/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.2.2+4
+
+* Update version of json_annotation, resolve conflicts with other packages e.g. flutter_tools from sdk.
+
```
Comment (cyanglaz) [inline packages/in_app_purchase/CHANGELOG.md]: maybe only:
`Update version of json_annotation to ^3.0.0 and json_serializable to ^3.2.0.` is enough?
cc @mklim

Comment (mklim) [reply packages/in_app_purchase/CHANGELOG.md]: I think either line is fine here. It's useful to know the exact version number and why we wanted to update it.

Comment (includecmath) [reply packages/in_app_purchase/CHANGELOG.md]: Ok, minor edits on the CHANGELOG.md

**Comment by includecmath** [comment]
What's up with CI?

**Comment by mklim** [comment]
> What's up with CI?

It's using the latest version of `flutter_plugin_tools`, but it looks like this branch is behind master so it's missing the configuration option that we need to pass to the latest version. If you merge in master and push again it should be fixed.

**Comment by includecmath** [comment]
@mklim CI seems time out in multiple PR.

**Comment by includecmath** [comment]
@mklim , can you please review this?

**Pull request merged**